rioxarray/rioxarray.py

```diff
@@ -203,12 +203,29 @@ def _make_coords(src_data_array, dst_affine, dst_width, dst_height, dst_crs):
     return add_xy_grid_meta(new_coords)
 
 
-def _make_dst_affine(src_data_array, src_crs, dst_crs, dst_resolution=None):
+def _make_dst_affine(
+    src_data_array, src_crs, dst_crs, dst_resolution=None, dst_shape=None
+):
     """Determine the affine of the new projected `xarray.DataArray`"""
     src_bounds = src_data_array.rio.bounds()
     src_width, src_height = src_data_array.rio.shape
+    dst_height, dst_width = dst_shape if dst_shape is not None else (None, None)
+    resolution_or_width_height = {
```

Comment:
Should `shape` be `(dst_width, dst_height)`, to conform to the source shape?

Reply:
If you explicitly pass in `dst_width` `dst_height`, rasterio will do the validation for you:
https://github.com/mapbox/rasterio/blob/ce00d482faa24068f41164b79901eeec31777263/rasterio/warp.py#L457-L467

Reply:
> Should shape be (dst_width, dst_height), to conform to the source shape?

That is actually a bug I think - those should be the other way around.

Reply:
> should we explicitly pass dst_width and dst_height?

I am fine either way. I like `shape` but `dst_width` and `dst_height` are used by `rasterio` and make the code simpler as you just pass in the values.

Reply:
I was also thinking that rioxarray offers a somewhat higher-level API than rasterio, and more similar to NumPy. That's why I thought that `(dst_height, dst_width)` is really the shape of the array. So I'm fine with keeping things like that, if you tell me that the source shape should also be `(src_height, src_width)`.

Reply:
Sounds good, then I think it should be good to go as is.